pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -160,21 +159,7 @@ func (pl *VolumeZone) getPVbyPod(ctx context.Context, pod *v1.Pod) ([]pvTopology
 		if s := getErrorAsStatus(err); !s.IsSuccess() {
 			return nil, s
 		}
-
-		for _, key := range topologyLabels {
-			if value, ok := pv.ObjectMeta.Labels[key]; ok {
-				volumeVSet, err := volumehelpers.LabelZonesToSet(value)
-				if err != nil {
-					logger.Info("Failed to parse label, ignoring the label", "label", fmt.Sprintf("%s:%s", key, value), "err", err)
-					continue
-				}
-				podPVTopologies = append(podPVTopologies, pvTopology{
-					pvName: pv.Name,
-					key:    key,
-					values: sets.Set[string](volumeVSet),
-				})
-			}
-		}
+		podPVTopologies = pl.getPVTopologyFromPV(logger, pv, podPVTopologies)
```

Comment:
This should be `podPVTopologies = append(podPVTopologies,pl.getPVTopologyFromPV(logger, pv, podPVTopologies))` right? I'm surprised the tests didn't fail🤔️

Reply:
No, I wrote `append` in `getPVTopologyFromPV` function (I might need to rename this function)
Because it is possible that pod has many labels, I have to call `append` in this function.

```
func (pl *VolumeZone) getPVTopologyFromPV(logger klog.Logger, pv *v1.PersistentVolume, podPVTopologies []pvTopology) []pvTopology {
	for _, key := range topologyLabels {
		if value, ok := pv.ObjectMeta.Labels[key]; ok {
			volumeVSet, err := volumehelpers.LabelZonesToSet(value)
			if err != nil {
				logger.Info("Failed to parse label, ignoring the label", "label", fmt.Sprintf("%s:%s", key, value), "err", err)
				continue
			}
			podPVTopologies = append(podPVTopologies, pvTopology{
				pvName: pv.Name,
				key:    key,
				values: sets.Set[string](volumeVSet),
			})
		}
	}
	return podPVTopologies
```

Reply:
Renamed the name of this function

Reply:
Ah, I missed the third param. In this case it's fine. But seems a little weird though.

Reply:
`But seems a little weird though.`

Gotha, I changed the code. I used `append` outside of `getPVTopologyFromPV`. It is better than previous implementation.